### PR TITLE
Turn on desktop and web integration tests for beta and stable

### PR DESCRIPTION
**Pull request by jmagman**
Make the branch restrictions on desktop and web integration tests match the enabled state in features.dart.

#45455
#68656
#68657
#68658

### CODE REVIEW

File: dev/bots/test.dart [hunk 1]
```diff
@@ -353,19 +332,18 @@ Future<void> _runBuildTests() async {
...
-        ),
-        // Should not fail to compile with dart:io.
-          () =>  _flutterBuildDart2js(
-          path.join('dev', 'integration_tests', 'web_compile_tests'),
-          path.join('lib', 'dart_io_import.dart'),
-        ),
-      ],
+    ...<ShardRunner>[
+      // Web compilation tests.
+        () =>  _flutterBuildDart2js(
```
Comment (yjbanov) [inline dev/bots/test.dart:337]: nit: should this be indented back by 2 spaces?

File: dev/bots/test.dart [hunk 2]
```diff
@@ -353,19 +332,18 @@ Future<void> _runBuildTests() async {
...
+        path.join('dev', 'integration_tests', 'web'),
+        path.join('lib', 'main.dart'),
+      ),
+      // Should not fail to compile with dart:io.
+        () =>  _flutterBuildDart2js(
```
Comment (yjbanov) [inline dev/bots/test.dart:342]: And this

**Review by yjbanov — APPROVED** [review]
lgtm

File: dev/bots/test.dart (hunk 1 above)
Comment (jmagman) [reply dev/bots/test.dart]: 👍   Reformatted.

**Pull request merged**